halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/model/v1/node/NodeFilter.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2016 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.halyard.config.model.v1.node;
+
+/**
+ * A node filter can be used to inform an iterator which elements to return.
+ */
+public class NodeFilter extends NodeReference {
+  private static final String ANY = "*";
+
+  public static boolean matches(String a, String b) {
+    if (a.isEmpty() || b.isEmpty()) {
+      return false;
+    }
+
+    if (a.equals(ANY) || b.equals(ANY)) {
+      return true;
+    }
+
+    return a.equals(b);
+  }
+
+  public NodeFilter anyHalconfigFile() {
```

Comment:
Good point. I'm not married to the NodeFilter setup just yet, so this may change in the near future.